controllers/flinkcluster_reconciler.go

```diff
@@ -520,6 +538,56 @@ func (reconciler *ClusterReconciler) cancelFlinkJob(jobID string, takeSavepoint
 	return reconciler.flinkClient.StopJob(apiBaseURL, jobID)
 }
 
+// Trigger savepoint if it is possible, then return the savepoint status to update.
+// When savepoint was already triggered, return the current observed status.
+// If triggering savepoint is impossible or skipped or triggered savepoint was created, proceed to stop the job.
+func (reconciler *ClusterReconciler) cancelFlinkJobAsync(jobID string, takeSavepoint bool) (*v1beta1.SavepointStatus, error) {
+	var log = reconciler.log
+	var cluster = reconciler.observed.cluster
+	var observedSavepoint = reconciler.observed.cluster.Status.Savepoint
+	var savepointStatus *v1beta1.SavepointStatus
+	var err error
+	if takeSavepoint {
```

Comment:
Can we move this if condition under `case v1beta1.SavepointStateNotTriggered`? I feel it will be more readable, and if in case there are on-going savepoints, we wait for it to complete.

Reply:
Good idea! I've changed to skip new savepoint in the case of there is one in progress already.